### PR TITLE
Allow blank connections applications use map-layers widget 's  UserPreferences

### PR DESCRIPTION
**Pull request by mdastous-bentley**
Previous implementation of UserPrefences assumes an iModelId and ItwinId are defined in IModelApp.  Applications using Blank connections (i.e. Reality Data Viewer) wants to save preferences without the need of an iTwin context.

### CODE REVIEW

File: extensions/map-layers/src/test/UserPreferencesMock.ts [hunk 1]
```diff
@@ -39,27 +41,42 @@ export function setup() {
...
+    if (!arg.key)
+      return Array.from(blankPrefs.values());
+    returnVal = blankPrefs.get(arg.key);
+
     return returnVal;
   };
 
   const deleteStub = async (arg: PreferenceKeyArg & ITwinIdArg & TokenArg) => {
-    if (undefined === iModelPrefs || undefined === iTwinPrefs)
+    if (undefined === iModelPrefs || undefined === iTwinPrefs || undefined === blankPrefs)
```
Comment (calebmshafer) [inline extensions/map-layers/src/test/UserPreferencesMock.ts:55]: Will a blank connection not have an iTwin Id though to associate it back to a project or other context?

Comment (mdastous-bentley) [reply extensions/map-layers/src/test/UserPreferencesMock.ts:55]: @calebmshafer 
yes I think you are right, there will always  be itwin ID with a blank connection, but I thought we could allow not passing it anyway; if the hosting application doesn't need it (i.e. saving the settings in the local storage).  If you prefer, I can keep the iTwin id mandatory.

File: extensions/map-layers/src/MapLayerPreferences.ts [hunk 2]
```diff
@@ -91,7 +91,7 @@ export class MapLayerPreferences {
    * @param projectId
    * @param iModelId
    */
-  public static async replaceSource(oldSource: MapLayerSource, newSource: MapLayerSource, projectId: GuidString, iModelId: GuidString): Promise<void> {
+  public static async replaceSource(oldSource: MapLayerSource, newSource: MapLayerSource, projectId?: GuidString, iModelId?: GuidString): Promise<void> {
```
Comment (calebmshafer) [inline extensions/map-layers/src/MapLayerPreferences.ts:94]: Looks like I accidentally left this as `projectId` mind switching it to `iTwinId` to match `storeSource`?

Comment (mdastous-bentley) [reply extensions/map-layers/src/MapLayerPreferences.ts:94]: done.

File: extensions/map-layers/src/test/UserPreferencesMock.ts (hunk 1 above)
Comment (mdastous-bentley) [reply extensions/map-layers/src/test/UserPreferencesMock.ts]: ok made iTwinId mandatory.

Comment (mdastous-bentley) [reply extensions/map-layers/src/test/UserPreferencesMock.ts]: @calebmshafer Please review one more time.

File: extensions/map-layers/src/MapLayerPreferences.ts (hunk 2 above)
Comment (calebmshafer) [reply extensions/map-layers/src/MapLayerPreferences.ts:94]: was this change pushed?

File: extensions/map-layers/src/test/UserPreferencesMock.ts (hunk 1 above)
Comment (calebmshafer) [reply extensions/map-layers/src/test/UserPreferencesMock.ts]: @mdastous-bentley looks like the `replaceSource` method still has it as optional. See comment https://github.com/iTwin/itwinjs-core/pull/3256#discussion_r817113559

Comment (mdastous-bentley) [reply extensions/map-layers/src/test/UserPreferencesMock.ts]: @calebmshafer should be good now

File: extensions/map-layers/src/MapLayerPreferences.ts (hunk 2 above)
Comment (mdastous-bentley) [reply extensions/map-layers/src/MapLayerPreferences.ts]: @calebmshafer Sorry missed few, should be good now.

File: extensions/map-layers/src/test/UserPreferencesMock.ts (hunk 1 above)
Comment (calebmshafer) [reply extensions/map-layers/src/test/UserPreferencesMock.ts]: cool, looks good.

**Review by calebmshafer — APPROVED** [review]

File: extensions/map-layers/src/MapLayerPreferences.ts [hunk 3]
```diff
@@ -288,11 +288,11 @@ export class MapLayerPreferences {
   }
 
   /** Get all MapLayerSources from the user's preferences, iTwin setting and iModel settings.
-   * @param projectId id of the project
+   * @param iTwinId id of the iTwiin
```
Comment (MarcBedard8) [inline extensions/map-layers/src/MapLayerPreferences.ts:291]: Typo: iTwiin -> two "i"

**Review by MarcBedard8 — APPROVED** [review]

File: extensions/map-layers/src/MapLayerPreferences.ts (hunk 3 above)
Comment (aruniverse) [reply extensions/map-layers/src/MapLayerPreferences.ts:291]:
```suggestion
   * @param iTwinId id of the iTwin
```

**Review by bsteinbk — APPROVED** [review]

**Pull request merged**